File: backend/nodejs-backend/src/models/DDLItem.js
class DDLItem {
    constructor({
        id = -1,
        name,
        startTime,
        endTime,
        isCompleted = false,
        completeTime = "",
        note,
        isArchived = false,
        isStared = false,
        type = "task",
        habitCount = 0,
        habitTotalCount = 0,
        calendarEvent = -1,
        timeStamp = ""
    }) {
        this.id = id;
        this.name = name;
        this.startTime = startTime;
        this.endTime = endTime;
        this.isCompleted = isCompleted;
        this.completeTime = completeTime;
        this.note = note;
        this.isArchived = isArchived;
        this.isStared = isStared;
        this.type = type;
        this.habitCount = habitCount;
        this.habitTotalCount = habitTotalCount;
        this.calendarEvent = calendarEvent;
        this.timeStamp = timeStamp;
    }
}

module.exports = DDLItem;

File: frontend/lib/models/ddl_item.dart
import 'package:frontend/models/deadline_type.dart';

class DDLItem {
  DDLItem({
    required this.id,
    required this.name,
    required this.startTime,
    required this.endTime,
    this.isCompleted = false,
    this.completeTime = '',
    this.note = '',
    this.isArchived = false,
    this.isStared = false,
    this.type = DeadlineType.TASK,
    this.habitCount = 0,
  });

  final int id;
  final String name;
  final String startTime;
  final String endTime;
  final bool isCompleted;
  final String completeTime;
  final String note;
  final bool isArchived;
  final bool isStared;
  final DeadlineType type;
  final int habitCount;
}